File: CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraTreeList.Nodes.Operations;
using DevExpress.XtraTreeList.Columns;
using DevExpress.XtraEditors;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(int));
            tbl.Columns.Add("Number", typeof(int));
            tbl.Columns.Add("Date", typeof(DateTime));
            tbl.Columns.Add("ParentID", typeof(int));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { String.Format("Name{0}", i), i + 1, 3 - i, DateTime.Now.AddDays(i), i % 3 });
            return tbl;
        }
   

        public Form1()
        {
            InitializeComponent();
            treeList1.DataSource = CreateTable(30);
        }

        private void simpleButton1_Click(object sender, EventArgs e)
        {
            treeList1.ExpandAll();
            PrintAllNodes(treeList1);
        }

        private void PrintAllNodes(DevExpress.XtraTreeList.TreeList treeList)
        {
            treeList.NodesIterator.DoLocalOperation(new PrintNodesOperation(listBoxControl1), treeList.Nodes);
        }
    }

    public class PrintNodesOperation : TreeListOperation
    {

        private ListBoxControl _ListBox;
        public PrintNodesOperation(ListBoxControl listBox)
        {
            _ListBox = listBox;
            
        }

        public override void Execute(DevExpress.XtraTreeList.Nodes.TreeListNode node)
        {
            PrintNode(node);
        }

        private void PrintNode(DevExpress.XtraTreeList.Nodes.TreeListNode node)
        {
            string nodeAsString = NodeToString(node);
            _ListBox.Items.Add(nodeAsString);
        }
        private string NodeToString(DevExpress.XtraTreeList.Nodes.TreeListNode node)
        {
            string res = string.Empty;
            string indent = "    ";
            for (int i = 0; i < node.Level; i++)
            {
                res += indent;
            }
            if (node.HasChildren)
                res += "+";
            else
                res += indent;
            res += "|";
            foreach (TreeListColumn col in node.TreeList.VisibleColumns)
            {
                res += String.Format("{0};", node.GetDisplayText(col));
            }
            res += "|" + Environment.NewLine;
            return res;
        }
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraTreeList.Nodes.Operations
Imports DevExpress.XtraTreeList.Columns
Imports DevExpress.XtraEditors

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Private Function CreateTable(ByVal RowCount As Integer) As DataTable
			Dim tbl As New DataTable()
			tbl.Columns.Add("Name", GetType(String))
			tbl.Columns.Add("ID", GetType(Integer))
			tbl.Columns.Add("Number", GetType(Integer))
			tbl.Columns.Add("Date", GetType(DateTime))
			tbl.Columns.Add("ParentID", GetType(Integer))
			For i As Integer = 0 To RowCount - 1
				tbl.Rows.Add(New Object() { String.Format("Name{0}", i), i + 1, 3 - i, DateTime.Now.AddDays(i), i Mod 3 })
			Next i
			Return tbl
		End Function


		Public Sub New()
			InitializeComponent()
			treeList1.DataSource = CreateTable(30)
		End Sub

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			treeList1.ExpandAll()
			PrintAllNodes(treeList1)
		End Sub

		Private Sub PrintAllNodes(ByVal treeList As DevExpress.XtraTreeList.TreeList)
			treeList.NodesIterator.DoLocalOperation(New PrintNodesOperation(listBoxControl1), treeList.Nodes)
		End Sub
	End Class

	Public Class PrintNodesOperation
		Inherits TreeListOperation

		Private _ListBox As ListBoxControl
		Public Sub New(ByVal listBox As ListBoxControl)
			_ListBox = listBox

		End Sub

		Public Overrides Sub Execute(ByVal node As DevExpress.XtraTreeList.Nodes.TreeListNode)
			PrintNode(node)
		End Sub

		Private Sub PrintNode(ByVal node As DevExpress.XtraTreeList.Nodes.TreeListNode)
			Dim nodeAsString As String = NodeToString(node)
			_ListBox.Items.Add(nodeAsString)
		End Sub
		Private Function NodeToString(ByVal node As DevExpress.XtraTreeList.Nodes.TreeListNode) As String
			Dim res As String = String.Empty
			Dim indent As String = "    "
			For i As Integer = 0 To node.Level - 1
				res &= indent
			Next i
			If node.HasChildren Then
				res &= "+"
			Else
				res &= indent
			End If
			res &= "|"
			For Each col As TreeListColumn In node.TreeList.VisibleColumns
				res &= String.Format("{0};", node.GetDisplayText(col))
			Next col
			res &= "|" & Environment.NewLine
			Return res
		End Function
	End Class
End Namespace